modules/library/jdbc/src/test/java/org/geotools/jdbc/JDBCFeatureSourceOnlineTest.java

```diff
@@ -690,4 +693,20 @@ public void testMixedEncodeIn() throws Exception {
             fail("Unexpected dialect, supports basic or prepared, but was a : " + dialect);
         }
     }
+
+    /**
+     * Online tests for String functions along with Like operator
+     *
+     * @throws Exception
+     */
+    public void testStringFunction() throws Exception {
+
+        FilterFactory ff = dataStore.getFilterFactory();
+        Function function = ff.function("strToLowerCase", ff.property("stringProperty"));
+
+        // should hit the row where stringProperty starts with z (e.g zero)
+        PropertyIsLike likeWithStringFunction = ff.like(function, "z%", "%", "-", "\\", true);
+        assumeTrue(dataStore.getFilterCapabilities().fullySupports(likeWithStringFunction));
```

Comment:
Please put assume as the first line in the test, makes the precondition clearer.